aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteCommandFactory.java

```diff
@@ -380,17 +406,29 @@ public SqlCommand deleteFor(@NonNull ModelSchema modelSchema,
                 SqlKeyword.DELIMITER +
                 sqlPredicate +
                 ";";
-        final SQLiteStatement compiledDeleteStatement =
-                databaseConnectionHandle == null ?
-                null : databaseConnectionHandle.compileStatement(preparedDeleteStatement);
         return new SqlCommand(table.getName(),
                 preparedDeleteStatement,
-                Collections.emptyList(),
-                sqlPredicate.getBindings(),
-                compiledDeleteStatement
+                sqlPredicate.getBindings() // WHERE clause
         );
     }
 
+    // extract model field values to save in database
+    private List<Object> extractFieldValues(@NonNull Model model) throws DataStoreException {
```

Comment:
We have a number of variants on this same basic concept across the DataStore code base, it feels like. In future work, we could try to bring some of them back together.

Reply:
Agreed, we could DRY up this logic somehow.

This is also an expensive operation (~5 ms for one Blog model instance).   When an an item is received from the backend, the following transformations occur: 
```
String -> JsonObject -> Java model -> Map<String, Object> (for saving to SQLite)
```

If we switch to using `SerializedModel` for everything internally, I believe it would improve performance quite a bit, since we would only need to do this:

```
String -> JsonObject -> Map<String, Object>
```